### PR TITLE
Add npm weekly and npm yearly downloads badge

### LINKED ISSUE #939: NPM monthly and weekly downloads
Not sure when it started but looks like all of the npm total downloads badges are incorrectly showing as zero. Monthly downloads appear to still be working. I noticed this on my own project but a few examples of well known packages below:

https://img.shields.io/npm/dt/express.svg 
https://img.shields.io/npm/dt/angular.svg 
https://img.shields.io/npm/dt/mocha.svg 

https://img.shields.io/npm/dm/express.svg 
https://img.shields.io/npm/dm/angular.svg 
https://img.shields.io/npm/dm/mocha.svg

### PR DESCRIPTION
**Pull request by mesaugat**
closes #939

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1632,6 +1625,64 @@ cache(function (data, match, sendBadge, request) {
...
+      } else {
+        badgeData.colorscheme = 'brightgreen';
+      }
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
```
Comment (Daniel15) [inline server.js]: Most of this code is copy and pasted across all three... Could you please generalise the parts of the code that's reusable?

Comment (mesaugat) [inline server.js]: Sure

Comment (mesaugat) [inline server.js]: Any ideas on how to generalize? There will be too many things to pass down. Not liking it.

Comment (Daniel15) [inline server.js]: The only things that differ between the three badges seem to be:
 - Badge URL (`dw`, `dm` , `dy`)
 - NPM API URL (`last-week`, `last-month`, `last-year`)
 - Suffix on the badge text (`/week`, `/month`, `/year`)

Everything else is the same. Given that, here's one idea (untested but should work):
```js
function mapNpmDownloads(urlComponent, apiUri, badgeSuffix) {
  camp.route(new RegExp('^\/npm\/' + urlComponent + '\/(.*)\.(svg|png|gif|jpg|json)$'),
  cache(function (data, match, sendBadge, request) {
    // .....
  });
}

mapNpmDownloads('dm', 'last-month', '/month');
mapNpmDownloads('dw', 'last-week', '/week');
mapNpmDownloads('dy', 'last-year', '/year');
```

Comment (mesaugat) [inline server.js]: Got it. Thanks!

**Comment by paulmelnikow** [comment]
@Daniel15 Would you like the honors? Typically I'd squash something like this through the UI, reworking the commit message as needed. Since there are no automated tests of the badges, you should do some of your own testing as well. A minimal version could be filtering `try.html` to make sure the badges are generating right.

**Comment by Daniel15** [comment]
@paulmelnikow  Sure, I can do that tomorrow or Friday! I want to test it out thoroughly before merging 😃

**Comment by Daniel15** [comment]
Tested locally and confirmed it all works as expected!

**Pull request merged**